lit_gpt/utils.py

```diff
@@ -35,7 +35,10 @@ def num_parameters(module: nn.Module, requires_grad: Optional[bool] = None) -> i
         if requires_grad is None or p.requires_grad == requires_grad:
             if hasattr(p, "quant_state"):
                 # bitsandbytes 4bit layer support
-                total += math.prod(p.quant_state[1])
+                if hasattr(p.quant_state, "shape"):
+                    total += math.prod(p.quant_state.shape)
+                else:
+                    total += math.prod(p.quant_state[1])
```

Comment:
I would say for simplicity we can drop this check. Lightning/Fabric verifies that the correct version of BNB is used anyway.

Reply:
ok fixing